Sources/SwiftFusion/Core/MatrixN.swift

```diff
@@ -1,8 +1,106 @@
-public struct Matrix3: Differentiable & Equatable & KeyPathIterable {
+public struct Matrix3: Differentiable & Equatable & KeyPathIterable & EuclideanVectorN & AdditiveArithmetic {
```

Comment:
Why do we need EuclideanVectorN & AdditiveArithmetic ? Don't we just need `move(:along)`?

Reply:
EuclideanVectorN is a protocol constraint for `JacobianFactor`, `AdditiveArithmetic` helps implementing the protocol constraint for `+=` and `-=` etc.